metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1610,3 +1613,97 @@ def visit_join_conversion_events_node(self, node: JoinConversionEventsNode) -> S
+        assert metric_instance and order_by_instance and partition_by_instances, (
+            "Did not receive appropriate instances to render SQL for WindowReaggregationNode. Expected instances matching "
+            f"specs: {expected_specs}. Got: {parent_instance_set.as_tuple}."
+        )
+
+        # Pending DSI upgrade:
+        # sql_window_function = SqlWindowFunction[
+        #     self._metric_lookup.get_metric(
+        #         metric_instance.spec.reference
+        #     ).type_params.cumulative_type_params.period_agg.name
+        # ]
+        sql_window_function = SqlWindowFunction.FIRST_VALUE  # placeholder for now
+        order_by_args = []
+        if sql_window_function.requires_ordering:
+            order_by_args.append(
+                SqlWindowOrderByArgument(
+                    expr=SqlColumnReferenceExpression.from_table_and_column_names(
+                        table_alias=parent_data_set_alias,
+                        column_name=order_by_instance.associated_column.column_name,
+                    ),
+                )
+            )
+        metric_select_column = SqlSelectColumn(
+            expr=SqlWindowFunctionExpression(
+                sql_function=sql_window_function,
+                sql_function_args=[
+                    SqlColumnReferenceExpression.from_table_and_column_names(
+                        table_alias=parent_data_set_alias, column_name=metric_instance.associated_column.column_name
+                    )
+                ],
+                partition_by_args=[
+                    SqlColumnReferenceExpression.from_table_and_column_names(
+                        table_alias=parent_data_set_alias,
+                        column_name=partition_by_instance.associated_column.column_name,
+                    )
+                    for partition_by_instance in partition_by_instances
+                ],
+                order_by_args=order_by_args,
+            ),
+            column_alias=metric_instance.associated_column.column_name,
+        )
+
+        # Order by instance should not be included in the output dataset because it was not requested in the query.
+        output_instance_set = parent_instance_set.transform(
+            FilterElements(exclude_specs=InstanceSpecSet(time_dimension_specs=(order_by_instance.spec,)))
+        )
+
+        # Can't include window function in a group by, so we use a subquery and apply group by in the outer query.
+        subquery_select_columns = output_instance_set.transform(
+            FilterElements(exclude_specs=InstanceSpecSet(metric_specs=(metric_instance.spec,)))
+        ).transform(
+            CreateSelectColumnsForInstances(parent_data_set_alias, self._column_association_resolver)
+        ).as_tuple() + (
+            metric_select_column,
+        )
+        subquery = SqlSelectStatementNode(
+            description="",  # description included in outer query
```

Comment:
Huh. This may look odd, though, every subquery is typically labeled. Maybe we just put something like "Window function for cumulative re-aggregation" or something?